app/models/claim/cleaner.rb

```diff
@@ -0,0 +1,50 @@
+class Claim
+  class Cleaner
+    attr_reader :claims
+
+    PII_ATTRIBUTES_TO_DELETE = [
+      :first_name,
+      :middle_name,
+      :surname,
+      :date_of_birth,
+      :address_line_1,
+      :address_line_2,
+      :address_line_3,
+      :address_line_4,
+      :postcode,
+      :payroll_gender,
+      :national_insurance_number,
+      :bank_sort_code,
+      :bank_account_number,
+      :building_society_roll_number,
+      :verify_response,
+    ]
+
+    def initialize(claims)
+      claims.each do |claim|
+        safe_to_clean(claim)
+      end
+      @claims = claims
+    end
+
+    def delete_pii
+      Claim.where(id: @claims.pluck(:id)).update_all(attribute_values_to_set)
+    end
+
+    private
+
+    def attribute_values_to_set
+      PII_ATTRIBUTES_TO_DELETE.map { |attr| [attr, nil] }.to_h.merge(
+        pii_removed_at: Time.now
+      )
+    end
+
+    def safe_to_clean(claim)
```

Comment:
Do we need to do this, given the only claims we're passing into the class are rejected or paid?

Reply:
This was to act as a sanity check in case something inadvertently passed an unsafe claim in, given that it's a (very) destructive operation.